### PR TITLE
Add top level error boundary

### PR DESCRIPTION
**Pull request by mfix22**
Not sure high up the component tree you wanted to place the boundary, so I just put it at the very top. 

Let me know if you would like me to change the position, or add styling to the error message/link. Happy to do so! 👍 

Closes https://github.com/gaearon/overreacted.io/issues/149

### CODE REVIEW

File: src/components/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+class Layout extends React.Component {
```
Comment (gaearon) [inline src/components/ErrorBoundary.js]: Let's call this ErrorBoundary?

Comment (mfix22) [reply src/components/ErrorBoundary.js]: Oops! My bad 👍

**Comment by mfix22** [comment]
Also, would you rather just use an error-boundary component lib like [`eitherx`](https://www.npmjs.com/package/eitherx)?

**Comment by gaearon** [comment]
Should probably be inside the body?

**Pull request closed without merging**